rfcs/proposed/parallel_block_for_task_arena/README.md

```diff
@@ -0,0 +1,108 @@
+covering most cases within its duration.
+
+The default behavior of oneTBB with these changes does not affect performance when oneTBB is used
+as the single parallel runtime.<br>
+However, some cases where several runtimes are used together might be affected. For example, if an
+application builds a pipeline where oneTBB is used for one stage and OpenMP is used for a
+subsequent stage, there is a chance that oneTBB workers will interfere with OpenMP threads.
+This interference might result in slight oversubscription,
+which in turn might lead to underperformance.
+
+This problem can be resolved with an API that indicates when parallel computation is done,
+allowing worker threads to be released from the arena,
+essentially overriding the default block-time.<br>
+
+This problem can be considered from another angle. Essentially, if the user can indicate where
+parallel computation ends, they can also indicate where they start.
+
+<img src="parallel_block_introduction.png" width=800>
+
+With this approach, the user not only releases threads when necessary
+but also specifies a programmable block where worker threads should stick to the
+executing arena.
+
+## Proposal
+
+Let's consider the guarantees that an API for explicit parallel blocks can provides:
+* Start of parallel block:
+  * Indicates the point from which the scheduler can use a hint and stick threads to the arena.
+  * Serve as a warm-up hint to the scheduler, making some worker threads immediately available
+    at the start of the real computatin.
+* "Parallel block" itself:
+  * Scheduler can implement different busy-wait policies to retain threads in the arena.
+* End of parallel block:
+  * Indicates the point from which the scheduler can drop a hint
+    and unstick threads from the arena.
+  * Indicates that worker threads should ignore
+    the default block time (introduced by PR#1352) and leave.
+
+Start of parallel block:<br>
+The warm-up hint should have similar guarantees as `task_arena::enqueue` from a signal standpoint.
+Users should expect the scheduler will do its best to make some threads available in the arena.
+
+"Parallel block" itself:<br>
+The guarantee for retaining threads is a hint to the scheduler;
+thus, no real guarantee is provided. The scheduler can ignore the hint and
+move threads to another arena or to sleep if conditions are met.
+
+End of parallel block:<br>
+It can indicate that worker threads should ignore the default block time but
+if work was submitted immediately after the end of the parallel block,
+the default block time will be restored.
+
+But what if user would like to disable default block time entirely?<br>
+Because the heuristic of extended block time is unsuitable for the task submitted
+in unpredictable pattern and duration. In this case, there should be an API to disable
+the default block time in the arena entirely.
+
+```cpp
+class task_arena {
+    void indicate_start_of_parallel_block(bool do_warmup = false);
```

Comment:
How about

`void retain_threads();`
`void release_threads();`


`*_parallel_block` is misleading since even your example shows serial parts of the region.

Reply:
Hmmm, I think `retain_threads` and `release_threads` provides unnecessary guaranteed like the threads will be actually retained.   
Should it be something more relaxed?
Perhaps, `make_sticky` and `make_unsticky` suites better because we can set definition of `sticky`.

Reply:
Do think "sticky" could people think of thread-to-core affinity? Since constraints are used for affinity, I'm thinking the likelihood of confusion is low and so I'm ok with `make_sticky` and `make_unsticky`.

Reply:
Should not all this be about work rather than threads? Threads are the execution resources, that should not be exposed to the user, should they? I mean that is the original idea of the TBB library. Therefore, I suggest something like `expect_[more/less_]parallel_work` or `assume_[more/less_]parallelism` as a good level of a loose terminology what library should tend to "_think_" about user's code when this API is used.

Reply:
TBB exposes some level of "thread logic" with `observers`. I'm not sure whether this API should expose this logic too.
If we want to extend these functions with additional guarantees such as "warm-up" or "leave earlier"  perhaps we could not ignore threads completely.